packages/ra-ui-materialui/src/auth/Login.tsx

```diff
@@ -74,86 +62,72 @@ const styles = (theme: Theme) =>
  *        </Admin>
  *     );
  */
-class Login extends Component<
-    Props & WithStyles<typeof styles> & HtmlHTMLAttributes<HTMLDivElement>
-> {
-    theme = createMuiTheme(this.props.theme);
-    containerRef = React.createRef<HTMLDivElement>();
-    backgroundImageLoaded = false;
+const Login: React.SFC<Props & HtmlHTMLAttributes<HTMLDivElement>> = ({
+    theme,
+    className,
+    children,
+    staticContext,
+    backgroundImage,
+    ...rest
+}) => {
+    const containerRef = useRef<HTMLDivElement>();
+    const styles = useStyles({});
+    let backgroundImageLoaded = false;
 
-    updateBackgroundImage = () => {
-        if (!this.backgroundImageLoaded && this.containerRef.current) {
-            const { backgroundImage } = this.props;
-            this.containerRef.current.style.backgroundImage = `url(${backgroundImage})`;
-            this.backgroundImageLoaded = true;
+    const updateBackgroundImage = () => {
+        if (!backgroundImageLoaded && containerRef.current) {
+            containerRef.current.style.backgroundImage = `url(${backgroundImage})`;
+            backgroundImageLoaded = true;
         }
     };
 
     // Load background image asynchronously to speed up time to interactive
-    lazyLoadBackgroundImage() {
-        const { backgroundImage } = this.props;
-
+    const lazyLoadBackgroundImage = () => {
         if (backgroundImage) {
             const img = new Image();
-            img.onload = this.updateBackgroundImage;
+            img.onload = updateBackgroundImage;
             img.src = backgroundImage;
         }
-    }
-
-    componentDidMount() {
-        this.lazyLoadBackgroundImage();
-    }
+    };
 
-    componentDidUpdate() {
-        if (!this.backgroundImageLoaded) {
-            this.lazyLoadBackgroundImage();
+    useEffect(() => {
+        if (!backgroundImageLoaded) {
+            lazyLoadBackgroundImage();
         }
-    }
-
-    render() {
-        const {
-            backgroundImage,
-            classes,
-            className,
-            children,
-            staticContext,
-            ...rest
-        } = this.props;
-
-        return (
-            <ThemeProvider theme={this.theme}>
-                <div
-                    className={classnames(classes.main, className)}
-                    {...rest}
-                    ref={this.containerRef}
-                >
-                    <Card className={classes.card}>
-                        <div className={classes.avatar}>
-                            <Avatar className={classes.icon}>
-                                <LockIcon />
-                            </Avatar>
-                        </div>
-                        {children}
-                    </Card>
-                    <Notification />
-                </div>
-            </ThemeProvider>
-        );
-    }
-}
+    });
 
-const EnhancedLogin = withStyles(styles)(Login) as ComponentType<Props>;
+    return (
+        <ThemeProvider theme={createMuiTheme(theme)}>
```

Comment:
You should probably initialize the theme inside a `useMemo` before passing it to the `ThemeProvider`